src/lib/mina_base/util/mina_base_util.ml

```diff
@@ -17,3 +17,25 @@ let split_last_exn =
   function [] -> failwith "split_last: Empty list" | x :: xs -> go [] x xs
 
 let two_to_the i = Bignum_bigint.(pow (of_int 2) (of_int i))
+
+(* TODO: This shouldn't live here.. *)
```

Comment:
Good point. The function is only used once, in `zkapp_examples.ml`. I speculate that @mrmr1993 moved it here to be able to use it in other examples as well.

As far as I can see, the function is not useful as a general way to do async computation inside an `exists` callback, because it violates the expectations that we usually have on `exists` callbacks - that we can read the values of vars.

Since it falls short of a general utility I think I'll just move it into `zkapp_examples.ml` where it happens to be useful